### DISCUSSION TITLE
Point reduction for polylines to improve performance.

### DISCUSSION THREAD

**Discussion by KrishnaRSatish** [osmdroid/osmdroid#1980] [Q&A]
I'm currently facing performance issues in my Android app when trying to draw polylines. I'm dynamically updating a list of GeoPoints every second, adding a point only if the distance between consecutive points is 10m. However, as the list grows, the device experiences slowdowns.

As per other discussions  point reductions was suggested, but considering my application runs on lower end devices, the point reduction algorithm is taking several minutes to process. For example a polyline with the list size of 80k geopoints is taking upto 2 mins for reduction.Since it is a live trace of the user, I would like a better approach.

@spyhunter99  
Any insights, code snippets or approaches is appreciated.

**Comment by monsieurtanuki** [comment]
Hi @KrishnaRSatish!
If your user loops in 8-shape with 80k geopoints and you want all the points to be visible, you're in trouble.

I would:
* simplify the 80k depending on the zoom level (every 4 geopoints for instance)
* "limit" the visible polyline to a given geobound, and tell the user recomputing needs to be done when the user goes beyond

Good luck!

**Comment by spyhunter99** [comment]
Try the point reducer algorithm

**Reply by MKergall** [reply]
As mentioned by @spyhunter99 , use the Douglas-Peucker PointReducer (in osmdroid util package). 
But start using it **before** your list is already too big... 
For instance, use it on a section of 1.000 points (should be pretty fast), and add this now simplified section to the already simplified trace.
